code/modules/mob/living/carbon/superior_animal/roach/attack.dm

```diff
@@ -1,15 +1,20 @@
 /mob/living/carbon/superior_animal/roach/UnarmedAttack(var/atom/A, var/proximity)
 	if(isliving(A))
 		var/mob/living/L = A
-		if(ishuman(A))
-			var/mob/living/carbon/human/H = L
+		var/mob/living/carbon/human/H
+		if(ishuman(L))
+			H = L
+		if(H)
 			var/obj/item/weapon/reagent_containers/food/snacks/grown/howdoitameahorseinminecraft = H.get_active_hand()
 			if(istype(howdoitameahorseinminecraft))
 				if(try_tame(H, howdoitameahorseinminecraft))
 					return FALSE //If they manage to tame the roach, stop the attack
 		if(istype(L) && !L.weakened && prob(5))
-			L.Weaken(3)
-			L.visible_message(SPAN_DANGER("\the [src] knocks down \the [L]!"))
+			if(H && H.has_shield())
+				. = ..()
```

Comment:
You could leave it empty or invert the check. There's `. = ..()` at bottom of UnarmedAttack

Reply:
I will replace it with visible_massage.

Reply:
Done.